### PR TITLE
Label operator namespace as privileged, add tests

### LINKED ISSUE #27: Meet emerging K8s pod security requirements
While running subctl diagnose commands on an OCP 4.11 version, the following warnings were noticed. The command seems to run fine, but we have to tweak the Security Context to fix these warnings.

```
subctl diagnose all
Cluster "amagrawa-c1-411"
 ✓ Checking Submariner support for the Kubernetes version
 ✓ Kubernetes version "v1.23.3+69213f8" is supported
 
 ✓ Checking Submariner support for the CNI network plugin
 ✓ The detected CNI network plugin ("OpenShiftSDN") is supported
 
 ✓ Checking gateway connections
 ✓ All connections are established
 
 ✓ Checking Submariner pods
 ✓ All Submariner pods are up and running
 
 ✓ Non-Globalnet deployment detected - checking if cluster CIDRs overlap
 ✓ Clusters do not have overlapping CIDRs
 
⠈⡱ Checking Submariner support for the kube-proxy mode W0516 16:30:50.377740   38759 warnings.go:67] would violate PodSecurity "restricted:latest": non-default capabilities (container "query-iface-list" must not include "NET_ADMIN", "NET_RAW" in securityContext.capabilities.add), host namespaces (hostNetwork=true), allowPrivilegeEscalation != false (container "query-iface-list" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "query-iface-list" must set securityContext.capabilities.drop=["ALL"]; container "query-iface-list" must not include "NET_ADMIN", "NET_RAW" in securityContext.capabilities.add), runAsNonRoot != true (pod or container "query-iface-list" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "query-iface-list" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
 ✓ Checking Submariner support for the kube-proxy mode
 ✓ The kube-proxy mode is supported
 
⠈⠑ Checking the firewall configuration to determine if the metrics port (8080) is allowed W0516 16:30:58.959021   38759 warnings.go:67] would violate PodSecurity "restricted:latest": non-default capabilities (container "validate-sniffer" must not include "NET_ADMIN", "NET_RAW" in securityContext.capabilities.add), host namespaces (hostNetwork=true), allowPrivilegeEscalation != false (container "validate-sniffer" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "validate-sniffer" must set securityContext.capabilities.drop=["ALL"]; container "validate-sniffer" must not include "NET_ADMIN", "NET_RAW" in securityContext.capabilities.add), runAsNonRoot != true (pod or container "validate-sniffer" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "validate-sniffer" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
⢄⡱ Checking the firewall configuration to determine if the metrics port (8080) is allowed W0516 16:31:05.487814   38759 warnings.go:67] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "validate-client" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "validate-client" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "validate-client" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "validate-client" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
 ✓ Checking the firewall configuration to determine if the metrics port (8080) is allowed
 ✓ The firewall configuration allows metrics to be retrieved from Gateway nodes
 
⠊⠁ Checking the firewall configuration to determine if VXLAN traffic is allowed W0516 16:31:16.307616   38759 warnings.go:67] would violate PodSecurity "restricted:latest": non-default capabilities (container "validate-sniffer" must not include "NET_ADMIN", "NET_RAW" in securityContext.capabilities.add), host namespaces (hostNetwork=true), allowPrivilegeEscalation != false (container "validate-sniffer" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "validate-sniffer" must set securityContext.capabilities.drop=["ALL"]; container "validate-sniffer" must not include "NET_ADMIN", "NET_RAW" in securityContext.capabilities.add), runAsNonRoot != true (pod or container "validate-sniffer" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "validate-sniffer" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
⢄⡱ Checking the firewall configuration to determine if VXLAN traffic is allowed W0516 16:31:22.879381   38759 warnings.go:67] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "validate-client" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "validate-client" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "validate-client" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "validate-client" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
 ✓ Checking the firewall configuration to determine if VXLAN traffic is allowed
 ✓ The firewall configuration allows VXLAN traffic
 
 ✓ Globalnet is not installed - skipping
 
Skipping inter-cluster firewall check as it requires two kubeconfigs. Please run "subctl diagnose firewall inter-cluster" command manually.
```

### PR DESCRIPTION
**Pull request by dfarrell07**
This is required for OpenShift 4.11.
    
Also add the privileged label during upgrades.
    
It would be better if these were Go unit tests, but we don't have those
for subctl yet.

Relates-to: submariner-io/subctl#27
Signed-off-by: Daniel Farrell <dfarrell@redhat.com

### CODE REVIEW

**Comment by dfarrell07** [comment]
This will fix the timeout (als going to re-trigger): https://github.com/submariner-io/subctl/pull/106

This shows that the tests correctly fail without the fix: https://github.com/dfarrell07/subctl/pull/4

**Comment by dfarrell07** [comment]
Re-running again...

File: pkg/namespace/ensure.go [hunk 1]
```diff
@@ -22,17 +22,28 @@ import (
...
-	_, err := kubeClient.CoreV1().Namespaces().Create(context.TODO(), ns, v1meta.CreateOptions{})
+func Ensure(kubeClient kubernetes.Interface, namespace string, namespaceLabels map[string]string) (bool, error) {
+	ns := &v1.Namespace{ObjectMeta: v1meta.ObjectMeta{Name: namespace, Labels: namespaceLabels}}
+
+	_, err := util.CreateOrUpdate(context.TODO(), resource.ForNamespace(kubeClient), ns, func(existing runtime.Object) (runtime.Object,
+		error,
+	) {
+		ns := existing.(*v1.Namespace)
+		for k, v := range namespaceLabels {
+			ns.Labels[k] = v
```
Comment (tpantelis) [inline pkg/namespace/ensure.go:48]: We shouldn't assume `ns.Labels` is non-nil, ie

```Go
if ns.Labels == nil {
    ns.Labels = map[string]string{}
}

for k, v := range namespaceLabels { 
    ....
}
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

**Review by tpantelis — APPROVED** [review]

**Review by sridhargaddam — APPROVED** [review]

**Review by skitt — APPROVED** [review]

**Comment by dfarrell07** [comment]
There was consensus on the dev call today that this should be backported to release-0.12.

**Comment by dfarrell07** [comment]
> There was consensus on the dev call today that this should be backported to release-0.12.

This was a complex backport, as the code is in the submariner-operator repo in the release-0.12 branch and it depends on a backport in Admiral.

Admiral backport: https://github.com/submariner-io/admiral/pull/377 
Operator backport: https://github.com/submariner-io/submariner-operator/pull/2117

**Pull request merged**